frame/identity/src/lib.rs

```diff
@@ -678,10 +419,15 @@ pub mod pallet {
 			for s in old_ids.iter() {
 				<SuperOf<T>>::remove(s);
 			}
-			let ids = subs.into_iter().map(|(id, name)| {
-				<SuperOf<T>>::insert(&id, (sender.clone(), name));
-				id
-			}).collect::<Vec<_>>();
+			let ids: BoundedVec<_, _> = subs
+				.into_iter()
+				.map(|(id, name)| {
+					<SuperOf<T>>::insert(&id, (sender.clone(), name));
+					id
+				})
+				.collect::<Vec<_>>()
+				.try_into()
+				.map_err(|_| Error::<T>::TooManySubAccounts)?;
```

Comment:
This cannot fail after you call `remove` above. We have a rule in the runtime, no returning error after state has changed.

This also means we are missing a test which would show that a failaure here is a `noop`

Reply:
the solution may be as simple as switching the order of the storage modifications here

Reply:
actually you cannot place that `insert` call into this loop which may fail at some point either...

Reply:
Looks like this operation will always succeed no matter what, because the length of `subs` has already been checked to be always lower than `T::MaxSubAccounts::get()`. The entire purpose of the `map` here is just to do an `insert` for `SuperOf`. Will rewrite this to make it more clear.